server/controllers/users.js

```diff
@@ -18,18 +18,33 @@ module.exports = {
     user.name = { firstname: req.body.firstname, lastname: req.body.lastname };
     user.email = req.body.email;
     user.password = req.body.password;
+    if (req.body.email === process.env.ADMIN_EMAIL) {
+      user.role = 'admin';
+    }
 
     user.save((err) => {
+      let error;
       if (err) {
         if (err.code === 11000) {
-          res.status(409).send({ message: 'Duplicate user entry.' });
+          error = err.message.match(/index:(.*?)_/)[1].trim();
+          res.status(409).send({ error: `Duplicate ${error}.` });
         } else {
-          res.status(400).send({ message: 'Error creating user.' });
+          if (!req.body.username) error = err.errors.username.message;
+          if (!req.body.email) error = err.errors.email.message;
+          if (!req.body.password) error = err.errors.password.message;
+          if (!req.body.firstname) {
+            error = err.errors['name.firstname'].message.replace(/name\./gi, '');
+          }
+          if (!req.body.lastname) {
+            error = err.errors['name.lastname'].message.replace(/name\./gi, '');
+          }
+          res.status(400).send({ error: `Error creating user: ${error}` });
         }
       } else {
         const userData = {
           _id: user._id,
           email: user.email,
+          role: user.role,
         };
```

Comment:
@andela-kakpobome the closing curly brace on line 48 is closing the userData object, so it's not really off